src/codegen/function.rs

```diff
@@ -406,7 +408,10 @@ pub fn body_chunk_futures(env: &Env, analysis: &analysis::functions::Info) -> St
         } else {
             let c_par = &analysis.parameters.c_parameters[par.ind_c];
 
-            if c_par.ref_mode != RefMode::None {
+            if *c_par.nullable {
+                try!(writeln!(body, "         {}.as_ref().map(::std::borrow::Borrow::borrow),",
```

Comment:
Please use '\t' here if it really needed

Reply:
and change next lines too

Reply:
I just took back the old code but fine with me!